utils/field.h

```diff
@@ -4,6 +4,8 @@
 #include "utils/fstack.h"
 #include "utils/list.h"
 
+#define ADDR_IN_48_BITS(addr) ((addr) & 0xffffffffffffUL)
```

Comment:
I guess the `UL` suffix is not enough for 32 bit systems.  Please make it `uint64_t` for sure.  Also I prefer having it as a (static inline) function over a macro (like `uint64_t effective_addr(uint64_t addr)`?)